packages/core/docs/guide/composition-api.md

```diff
@@ -0,0 +1,189 @@
+# Composition API
+
+> If you already have an experience with Vue Composition API, you can skip this section and start using [Vue Storefront Composables](#what-are-vue-storefront-composables)
+
+Composition API is a new way of abstracting and reusing the logic added in Vue 3.0. It allows you to create and observe a reactive state both inside the Vue component and outside of it as a standalone function.
+
+Let's try to build functionality for submitting a form with two fields: user name and password.
+
+```html
+<template>
+  <form>
+    <input type="text" placeholder="Username" />
+    <input type="password" placeholder="Password" />
+    <button type="submit">Submit</button>
+  </form>
+</template>
+```
+
+In the Vue component, Composition API methods should be called inside the new component option called `setup`:
+
+```html
+<script>
+  export default {
+    setup() {
+      // here we will start using Composition API
+    },
+    data() {
+      return {
+        /* ... */
+      };
+    },
+    methods: {
+      /* ... */
+    }
+  };
+</script>
+```
+
+First of all, we need two reactive values that will be bound to form fields. They can be created with Vue `ref` method:
+
+```js
+import { ref } from 'vue';
+
+export default {
+  setup() {
+    const username = ref('');
+    const password = ref('');
+
+    return { username, password };
+  }
+};
+```
+
+The argument passed to `ref` is an _initial value_ of our reactive state properties. Under the hood, `ref` creates an object with a single property `value`:
+
+```js
+setup() {
+  const username = ref('');
+  const password = ref('');
+
+  console.log(username.value) // => ''
+
+  return { username, password };
+}
+```
+
+After we returned reactive properties from the `setup`, they become available in Vue component options (such as `data` or `methods`) and in component template. `refs` returned from setup are automatically unwrapped - this means you don't access the `.value` anymore:
+
+```html
+<template>
+  <form @submit="submitForm">
+    <input v-model="username" type="text" placeholder="Username" />
+    <input v-model="password" type="password" placeholder="Password" />
+    <button type="submit">Submit</button>
+  </form>
+</template>
+
+<script>
+  import { ref } from 'vue';
+
+  export default {
+    setup() {
+      const username = ref('');
+      const password = ref('');
+
+      return { username, password };
+    }
+    methods: {
+      submitForm() {
+        console.log(this.username)
+      }
+    }
+  };
+</script>
+```
+
+Methods can be also created inside the `setup` option:
+
+```js{5-11}
```

Comment:
I don't know if my math is right, but shouldn't it be `{6-11}`?

Reply:
Right, thx!